plugins/include/amxmisc.inc

```diff
@@ -337,20 +337,18 @@ stock show_activity_custom(id, const name[], const fmt[], any:...)
 	if(!activity_init())
 		return;
 
-	static buffer[192];
+	new buffer[192];
 	vformat(buffer, charsmax(buffer), fmt, 4);
 
 	if(!id)
 	{
-		static players[MAX_PLAYERS], pnum;
+		new players[MAX_PLAYERS], pnum;
 		get_players(players, pnum);
 
 		if(!pnum)
 			return
 
-		static player;
-
-		for(new i; i < pnum; i++)
+		for(new i, player; i < pnum; i++)
 		{
 			player = players[i];
```

Comment:
Hm? It is `new`.

Reply:
@OciXCrom I mean inside for-loop block.

Reply:
@WPMGPRoSToTeMa It's declared inside the for-loop body.

>for(new i, player; i < pnum; i++)

I think you're viewing the old version of the code.

Reply:
@OciXCrom I mean its declaration should be moved to its first assignment.

Reply:
@OciXCrom inside for-loop iteration block. The scope of variable should be as small as possible.
```Pawn
for (new i = 0; i < pnum; i++)
{
	new player = players[i];
```

Reply:
@OciXCrom why not?

Reply:
@WPMGPRoSToTeMa Because everyone always says that it's bad for the code. The variable will be created 32 times (in a full server), right now it's declared only once and the scope is the same AFAIK. Even @Arkshine has said this in the forum.

Reply:
It just doesn't matter. Narrowing the scope is better than saving 1 instruction per iteration. A smart compiler (not amxxpc) would optimize that out anyway.
You can also declare it in the first "part" (before the `;`) of the for loop, or just create another scope.

Reply:
You fool, never quote me!

Reply:
@rsKliPPy That's exactly what I did. I declared the variable before the ";" in the loop like I always do, so I don't see a problem. I've always been told that declaring them in the actual loop is bad.

Reply:
@OciXCrom look the difference https://www.diffchecker.com/PUgvCnnc.

Reply:
@OciXCrom did you look into this assembly? Do you have any questions/thoughts?

Reply:
@WPMGPRoSToTeMa I can't say that I understand much what that even means. Maybe you can explain? If it's not something extremely important, let's just keep it declared in the for() body for the sake of everyone knowing that that is the better way to do it.